src/index.ts

```diff
@@ -1,55 +1,65 @@
 import { isAnyArray } from 'is-any-array';
 
+import checkArrayLength, { type NumberArray } from './checkArrayLength';
+
 export { default as maybeToPrecision } from './maybeToPrecision';
-export { default as checkArrayLength } from './checkArrayLength';
 
+export interface RegressionScore {
+  r: number;
+  r2: number;
+  chi2: number;
+  rmsd: number;
+}
 export default class BaseRegression {
   constructor() {
     if (new.target === BaseRegression) {
       throw new Error('BaseRegression must be subclassed');
     }
   }
 
-  predict(x) {
+  predict(x: number): number;
+  predict(x: number[]): number[];
+  predict(x: number | number[]) {
```

Comment:
why not NumberArray here?

Reply:
NumberArray is in cheminto-types: https://github.com/cheminfo/cheminfo-types/blob/main/src/core/NumberArray.d.ts
This package (cheminfo-types) is where we place also the share types definition.